pkg/ext-proc/scheduling/scheduler.go

```diff
@@ -42,13 +45,39 @@ var (
 		},
 	}
 
+	// queueAndKVCacheFilter applies least queue followed by least KV Cache filter
+	queueAndKVCacheFilter = &filter{
+		name:   "least queuing",
+		filter: leastQueuingFilterFunc,
+		nextOnSuccessOrFailure: &filter{
+			name:   "least KV cache percent",
+			filter: leastKVCacheFilterFunc,
+		},
+	}
+
+	lowLatencyFilter = &filter{
+		name:   "low queueing filter",
+		filter: toFilterFunc((lowQueueingPodPredicate)),
+		nextOnSuccess: &filter{
+			name:          "affinity LoRA",
+			filter:        toFilterFunc(loRAAffinityPredicate),
```

Comment:
why not use `lowLoRACostPredicate` with `nextOnSuccessOrFailure: queueAndKVCacheFilter` instead of doing `loRAAffinityPredicate` and `canAcceptNewLoraPredicate` separately?

Reply:
lowLoRACostPredicate picks both pods with canAcceptNewLoraPredicate and loRAAffinityPredicate, For stronger affinity we want to pick only pods with loRAAffinityPredicate and if no such pod is present only then pick canAcceptNewLoraPredicate

Reply:
Why not do that for the other branch too then?

Reply:
The lowLoRACostPredicate ensures weak affinity by spreading the load of a LoRA adapter across multiple pods, avoiding "pinning" all requests to a single pod. This gave good performance in our initial benchmarking results in the scenario where # of lora slots < # of lora adapters.

Reply:
Can we document this reasoning please?

Reply:
I added a comment to lowLoRACostPredicate with the reasoning, like we have in leastKVCacheFilterFunc.